### PR TITLE
#758: Cleanup different singletonian dialects

### PR DESCRIPTION
**Pull request by bonkey**
Well, that's the concept, too bad I got this on the first run:

```
error: -[AtlasUITests.CheckoutSummaryEditProductStackViewTests 
  testOneSizeInitialSizeWhileLoggedIn] : expected to eventually not be nil, got <nil>

error: -[AtlasUITests.CheckoutSummaryEditProductStackViewTests
  testSelectInitialSizeWhileLoggedIn] : expected to eventually not be nil, got <nil>
```

### CODE REVIEW

**Comment by bonkey** [comment]
:+1:

File: AtlasUI/AtlasUI/AtlasUI.swift [hunk 1]
```diff
@@ -53,48 +44,40 @@ final public class AtlasUI {
...
 
-    /// `presentCheckout` is used to present the `AtlasUI` over the given controller
-    /// It can be called only once at a time as it depends internally on singleton objects
-    ///
-    /// - Parameters:
-    ///   - viewController: The controller in which `AtlasUI` will be presented over it
-    ///   - sku: The SKU for the item that the user want to buy
-    /// - Throws: `AtlasUI.Error.notInitialized` when `AtlasUI` is not finished `AtlasUI.configure(options:completion:)`
     public func presentCheckout(onViewController viewController: UIViewController, for sku: ConfigSKU) throws {
-        guard let _ = AtlasAPI.shared else {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasUI.swift]: no more throwing?

File: AtlasUI/AtlasUI/AtlasUI.swift [hunk 2]
```diff
@@ -53,48 +44,40 @@ final public class AtlasUI {
         }
     }
 
-    /// `presentCheckout` is used to present the `AtlasUI` over the given controller
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasUI.swift]: my documentations 😭

File: AtlasUI/AtlasUI/AtlasUI.swift [hunk 3]
```diff
@@ -53,48 +44,40 @@ final public class AtlasUI {
...
     }
 
-    func provide<T>() throws -> T {
-        return try injector.provide()
+}
+
+extension AtlasUI {
+
+    static func fromPresented
+        () throws -> AtlasUI {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasUI.swift]: swiftmat? 👹

File: AtlasUI/AtlasUI/AtlasUIViewController.swift [hunk 4]
```diff
@@ -7,13 +7,14 @@ import AtlasSDK
 
 class AtlasUIViewController: UIViewController {
 
-    static var shared: AtlasUIViewController? {
-        return try? AtlasUI.shared().provide()
-    }
+    fileprivate(set) static var presented: AtlasUIViewController?
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/AtlasUIViewController.swift]: weak?
no be cleared when the view is dismissed

**Comment by hani-ibrahim** [comment]
👍

**Comment by bonkey** [comment]
👍

**Pull request merged**